File: HandSock-Server/src/main/java/com/server/handsock/clients/service/ClientChannelService.java
package com.server.handsock.clients.service;

import com.baomidou.mybatisplus.core.conditions.query.QueryWrapper;
import com.server.handsock.console.ConsolePrints;
import com.server.handsock.console.HandleResults;
import com.server.handsock.services.HistoryService;
import com.server.handsock.clients.dao.ClientChannelDao;
import com.server.handsock.clients.mod.ClientChannelModel;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.*;

@Service
public class ClientChannelService {

    private final HistoryService historyService;
    private final ClientChannelDao clientChannelDao;
    private final ClientChatService clientChatService;
    private final HandleResults handleResults = new HandleResults();

    @Autowired
    public ClientChannelService(ClientChannelDao clientChannelDao, HistoryService historyService, ClientChatService clientChatService) {
        this.historyService = historyService;
        this.clientChannelDao = clientChannelDao;
        this.clientChatService = clientChatService;
    }

    public Map<String, Object> searchAllGroup() {
        try {
            List<ClientChannelModel> OPChanModelList = clientChannelDao.selectList(new QueryWrapper<ClientChannelModel>().eq("active", 1));

            List<Map<String, Object>> history = new ArrayList<>();
            for (ClientChannelModel OPChanModel : OPChanModelList) {
                history.add(new HashMap<>(){{
                    put("gid", OPChanModel.getGid());
                    put("latest", clientChatService.searchHistoryById(historyService.queryGroupLeastHistory(OPChanModel.getGid())));
                }});
            }
            Map<String, Object> result = new HandleResults().handleResultByCode(200, OPChanModelList, "获取成功");
            result.put("latest", history);
            return result;
        } catch (Exception e) {
            new ConsolePrints().printErrorLog(e);
            return handleResults.handleResultByCode(500, null, "服务端异常");
        }
    }

    public Map<String, Object> searchGroupByUid(long gid) {
        try {
            ClientChannelModel groupSelect;
            if (gid == 0) {
                groupSelect = clientChannelDao.selectOne(new QueryWrapper<ClientChannelModel>().eq("home", 1));
            } else groupSelect = clientChannelDao.selectOne(new QueryWrapper<ClientChannelModel>().eq("gid", gid));
            if (groupSelect == null) {
                return handleResults.handleResultByCode(404, null, "未找到频道");
            } else return handleResults.handleResultByCode(200, groupSelect, "获取成功");
        } catch (Exception e) {
            new ConsolePrints().printErrorLog(e);
            return handleResults.handleResultByCode(500, null, "服务端异常");
        }
    }

    public Boolean getChanOpenStatus(Long gid) {
        try {
            Integer status = clientChannelDao.selectOne(new QueryWrapper<ClientChannelModel>().eq("gid", gid)).getOpen();
            return Objects.equals(status, 1);
        } catch (Exception e) {
            new ConsolePrints().printSuccessLog(e);
            return false;
        }
    }
}

File: HandSock-Server/src/main/java/com/server/handsock/clients/service/ClientChannelService.kt
package com.server.handsock.clients.service

import com.baomidou.mybatisplus.core.conditions.query.QueryWrapper
import com.server.handsock.clients.dao.ClientChannelDao
import com.server.handsock.clients.mod.ClientChannelModel
import com.server.handsock.utils.HandUtils
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class ClientChannelService @Autowired constructor(private val clientChannelDao: ClientChannelDao) {
    fun searchAllGroup(): Map<String, Any> {
        try {
            val clientChannelModelList = clientChannelDao.selectList(QueryWrapper<ClientChannelModel>().eq("active", 1))

            val history: MutableList<Map<String, Any>> = ArrayList()
            for (clientChannelModel in clientChannelModelList) {
                history.add(object : HashMap<String, Any>() {
                    init {
                        clientChannelModel.gid?.let { put("gid", it) }
                    }
                })
            }
            val result = HandUtils.handleResultByCode(200, clientChannelModelList, "获取成功").apply {
                "latest" to history
            }
            return result
        } catch (e: Exception) {
            return HandUtils.printErrorLog(e)
        }
    }

    fun searchGroupByGid(gid: Long): Map<String, Any> {
        try {
            val groupSelect = if (gid == 0L) {
                clientChannelDao.selectOne(QueryWrapper<ClientChannelModel>().eq("home", 1))
            } else clientChannelDao.selectOne(QueryWrapper<ClientChannelModel>().eq("gid", gid))
            return if (groupSelect == null) {
                HandUtils.handleResultByCode(404, null, "未找到频道")
            } else HandUtils.handleResultByCode(200, groupSelect, "获取成功")
        } catch (e: Exception) {
            return HandUtils.printErrorLog(e)
        }
    }

    fun getChanOpenStatus(gid: Long?): Boolean {
        try {
            val status = clientChannelDao.selectOne(QueryWrapper<ClientChannelModel>().eq("gid", gid)).open
            return status == 1
        } catch (e: Exception) {
            HandUtils.printErrorLog(e)
            return false
        }
    }
}
